### PR TITLE
Add `SEMICOLON_IN_EXPRESSIONS_FROM_MACROS` to future-incompat report

### PR DESCRIPTION
**Pull request by Aaron1011**
See https://github.com/rust-lang/rust/issues/79813 for a discussion of this lint. This has been warn-by-default for over a year, so adding it to the future-incompat report should help to find libraries that haven't yet updated.

### CODE REVIEW

**Review by davidtwco — APPROVED** [review]

**Comment by davidtwco** [comment]
r=me, not sure if this needs sign-off from t-compiler or t-lang or anything like that

**Comment by pnkfelix** [comment]
I honestly cannot remember the protocol for deciding when a lint should be added to the future-incompat report.

The template says: 
> At the beginning of each 6-week release cycle, the Rust compiler team
> will review the set of outstanding future compatibility warnings and
> nominate some of them for **Final Comment Period**.

The lang team already approved work being done here under https://github.com/rust-lang/lang-team/issues/70

but that approval did include the caveat that the authors "should I-nominate the actual implementation PR once it's ready for final approval by lang team."

but then subsequently, https://github.com/rust-lang/rust/pull/87385#issuecomment-887719443 shows that the lang team *did* approve upgrading the lint to warn-by-default.

@Aaron1011 , just to be clear: You haven't yet made the lint active for a macro defined in a different crate from where it is used, right?

In any case, I think we are in a good place to nominate for compiler team FCP, as described in the future-incompat template.

(I'm also tagging this with I-lang-nominated to make sure lang team has chance to put eyes on it. But I'm not treating lang team discussion as a blocker for the compiler team to move forward with an FCP here, because I think the lang teams prior approval should suffice here.)

@rfcbot fcp merge

**Comment by joshtriplett** [comment]
Thanks for the heads-up, and @rust-lang/lang members can review at their discretion.

**Comment by Aaron1011** [comment]
cc @davidtwco @michaelwoerister @nagisa @nikomatsakis for checkboxes

**Pull request merged**
